src/stdplus/eint.hpp

```diff
@@ -18,24 +19,49 @@ namespace stdplus {
 namespace eint {
 
 /// Calculate (a-b) in safe way (see documentation and WARNINGS about this namespace)
-/// @TODO now assumes T is UNSIGNED type!
 /// @TODO can variable "b" be silently converted to type T here, including narrowing? if yes then template on T1,T2 and
 /// ...check if b can be properly expanded to T1 ?
 template <typename T>
-T eint_minus(T a, T b) {
-	if (a<b) {
+T eint_minus(T a, T b, typename std::enable_if<std::is_integral<T>::value>::type* = 0) {
+	static_assert(std::numeric_limits<T>::min()<=0, "Integral type not include 0");
+	static_assert(std::numeric_limits<T>::max()>=0, "Integral type not include 0"); // it is normal integer with 0 value
+	bool is_overflow=false;
+	if (b<0)
+	{
+		if (std::numeric_limits<T>::max()+b < a) is_overflow = true;
+	}
+	if (b>0)
+	{
+		if (std::numeric_limits<T>::min()+b > a) is_overflow = true;
+	}
+	if (is_overflow) {
 		std::ostringstream oss; oss<<"Math error when calculating (" << a << " - " << b << ") for T="<<typeid(T).name();
 		throw std::range_error(oss.str());
 	}
-	T c = a - b;
-	return c;
+	return a-b;
 }
 
-// TODO
+/// Calculate (a+b) in safe way (see documentation and WARNINGS about this namespace)
+/// @TODO can variable "b" be silently converted to type T here, including narrowing? if yes then template on T1,T2 and
+/// ...check if b can be properly expanded to T1 ?
 template <typename T>
-T eint_plus(T a, T b) {
-	T c = a + b;
-	return c;
+T eint_plus(T a, T b, typename std::enable_if<std::is_integral<T>::value>::type* = 0) {
+	static_assert(std::numeric_limits<T>::min()<=0, "Integral type not include 0");
+	static_assert(std::numeric_limits<T>::max()>=0, "Integral type not include 0"); // it is normal integer with 0 value
+	bool is_overflow=false;
+	if (a>0)
+	{
+		if (std::numeric_limits<T>::max()-a < b) is_overflow = true;
+	}
+	if (a<0)
+	{
+		if (std::numeric_limits<T>::min()-a > b) is_overflow = true;
```

Comment:
it will be works only for unsigned integers [basic.fundamental] 3.9.1/4, for signed this is UB [expr] 5/4